src/api/db/migrate/20170427102943_add_references_to_bs_request_action.rb

```diff
@@ -0,0 +1,34 @@
+class AddReferencesToBsRequestAction < ActiveRecord::Migration[5.0]
+  def up
+    start = Time.now
+    add_reference(:bs_request_actions, :source, polymorphic: true)
+    add_reference(:bs_request_actions, :target, polymorphic: true)
+
+    target_projects = BsRequestAction.where("target_project IS NOT NULL AND target_package IS NULL").distinct(:target_project).pluck(:target_project)
+    source_projects = BsRequestAction.where("source_project IS NOT NULL AND source_package IS NULL").distinct(:source_project).pluck(:source_project)
+    tmp = (target_projects + source_projects).uniq
+
+    tmp.in_groups_of(10000) do |array|
+      Project.where(name: array).each do |project|
+        execute "UPDATE bs_request_actions SET source_id = #{project.id}, source_type = 'Project' WHERE source_project = '#{project.name}' AND source_package IS NULL"
+        execute "UPDATE bs_request_actions SET target_id = #{project.id}, target_type = 'Project' WHERE target_project = '#{project.name}' AND target_package IS NULL"
+      end
+    end
```

Comment:
Similar to the other migration, can we do this without having to load all the data into ruby? For example (also untested):
```sql
UPDATE bs_request_actions
INNER JOIN projects ON projects.name = bs_request_actions.target_project
SET bs_request_actions.target_id = projects.id, target_type = 'Project'
WHERE bs_request_actions.target_project IS NOT NULL AND bs_request_actions.target_package IS NULL
```